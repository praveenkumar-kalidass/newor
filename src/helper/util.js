import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
