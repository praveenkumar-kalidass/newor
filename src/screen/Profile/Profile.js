import React, { useMemo, useState } from 'react';
import { KeyboardAvoidingView, Platform, TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import {
  Center, Avatar, Flex, VStack, FlatList, Text, Box, Divider, Spinner,
} from 'native-base';
import Fontawesome from 'react-native-vector-icons/FontAwesome';
import { launchImageLibrary } from 'react-native-image-picker';

import COLOR from 'constant/color';
import useTranslation from 'translation/useTranslation';
import Translation from 'translation/Translation';
import useTheme from 'theme/useTheme';
import useUserApi from 'api/useUser';
import useUser from 'provider/User/useUser';
import ToastAlert from 'component/ToastAlert';
import TitleBar from 'component/TitleBar/TitleBar';
import withBackground from 'helper/withBackground';
import { Background4 } from 'component/Background';
import useError from 'hook/useError';
import { Container, EditBadge } from './Profile.style';

const Profile = () => {
  const [isSubmit, setIsSubmit] = useState(false);
  const { user, setUser } = useUser();
  const { updatePicture } = useUserApi();
  const { translate } = useTranslation();
  const theme = useTheme();
  const list = useMemo(() => [
    {
      label: translate('FIRST_NAME'),
      key: 'firstName',
    },
    {
      label: translate('LAST_NAME'),
      key: 'lastName',
    },
    {
      label: translate('EMAIL'),
      key: 'email',
    },
    {
      label: translate('MOBILE_NUMBER'),
      key: 'mobileNumber',
    },
  ], []);
  const { toast } = useError();
  const navigation = useNavigation();

  const changePicture = async () => {
    try {
      setIsSubmit(true);
      const result = await launchImageLibrary();
      if (result.didCancel) {
        setIsSubmit(false);
        return;
      }
      const { data } = await updatePicture(result.assets[0]);
      setUser((previous) => ({ ...previous, picture: data.picture }));
      setIsSubmit(false);
      toast.show({
        render: () => <ToastAlert status="success" message={translate('PICTURE_UPDATED')} />,
        placement: 'top',
      });
    } catch (error) {
      setIsSubmit(false);
      toast(error);
    }
  };

  return (
    <KeyboardAvoidingView flex={1} behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
      <Flex flex={1} direction="row" p="5" justify="center">
        <Container>
          <Box flexDirection="row" alignItems="center" mb={10} w="100%" justifyContent="space-between">
            <TitleBar color={theme.color.SECONDARY_100}>
              <Translation tkey="PROFILE" as={Text} bold fontSize={20} />
            </TitleBar>
            <TouchableOpacity onPress={navigation.goBack}>
              <Fontawesome color={theme.color.BACKGROUND_0} size={24} name="close" />
            </TouchableOpacity>
          </Box>
          <Center>
            <TouchableOpacity onPress={changePicture} testID="profile-update-picture">
              <EditBadge>
                <Fontawesome color={COLOR.LIGHT_BACKGROUND_100} size={20} name="pencil" />
              </EditBadge>
              <If condition={isSubmit}>
                <Avatar bg={COLOR.LIGHT_BACKGROUND_100} size="xl">
                  <Spinner color={COLOR.LIGHT_BACKGROUND_0} />
                </Avatar>
              </If>
              <If condition={!isSubmit}>
                <Avatar bg={COLOR.LIGHT_BACKGROUND_100} size="xl" source={{ uri: user.picture }}>
                  <Fontawesome size={40} name="user-secret" />
                </Avatar>
              </If>
            </TouchableOpacity>
          </Center>
          <Box my={5}>
            <FlatList
              data={list}
              renderItem={({ item }) => (
                <VStack>
                  <Text bold mb={1}>{item.label}</Text>
                  <Text fontSize={16}>{user[item.key]}</Text>
                  <Divider my={2} />
                </VStack>
              )}
              keyExtractor={(item) => item.key}
            />
          </Box>
        </Container>
      </Flex>
    </KeyboardAvoidingView>
  );
};

export default withBackground({ Component: Profile, Background: Background4 });
