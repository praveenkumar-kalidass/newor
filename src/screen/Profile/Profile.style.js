import styled from 'styled-components/native';
import { Badge, Box } from 'native-base';

export const Container = styled(Box).attrs(({ theme }) => ({
  borderRadius: 20,
  flex: 1,
  alignSelf: 'center',
  p: 5,
  bg: theme.color.MODAL,
  elevation: 1,
  shadowColor: theme.color.BACKGROUND_0,
  shadowOffset: { width: 0, height: 1 },
  shadowOpacity: 0.8,
  shadowRadius: 1,
}))``;

export const EditBadge = styled(Badge).attrs(({ theme }) => ({
  bg: theme.color.PRIMARY_100,
  rounded: 'full',
  mb: '-5',
  mr: '-5',
  zIndex: 1,
  variant: 'solid',
  alignSelf: 'flex-end',
}))``;
