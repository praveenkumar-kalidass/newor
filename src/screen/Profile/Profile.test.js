import React from 'react';
import { Platform } from 'react-native';
import { launchImageLibrary } from 'react-native-image-picker';

import { render, act, fireEvent } from 'test/util';
import Profile from './Profile';

jest.mock('provider/User/useUser', () => () => ({
  user: {
    firstName: 'Hello',
    lastName: 'World',
    email: 'helloworld@test.com',
    mobileNumber: '9876543210',
  },
  setUser: jest.fn(),
}));
jest.mock('react-native-image-picker', () => ({
  launchImageLibrary: jest.fn(),
}));
const mockUseUserApi = {
  updatePicture: jest.fn(),
};
jest.mock('api/useUser', () => () => mockUseUserApi);
const mockToast = { show: jest.fn() };
jest.mock('native-base', () => {
  const nativeBase = jest.requireActual('native-base');
  return {
    ...nativeBase,
    useToast: () => mockToast,
  };
});

describe('Profile', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const container = render(<Profile />);

    expect(container).toMatchSnapshot();
  });

  it('should update profile picture successfully', async () => {
    Platform.OS = 'ios';
    launchImageLibrary.mockResolvedValueOnce({ assets: [] });
    mockUseUserApi.updatePicture.mockResolvedValueOnce({ data: { picture: 'test.png' } });

    const { getByTestId } = render(<Profile />);

    await act(async () => {
      await fireEvent.press(getByTestId('profile-update-picture'));
    });

    expect(mockToast.show).toHaveBeenCalledTimes(1);
  });

  it('should not update profile picture', async () => {
    Platform.OS = 'android';
    launchImageLibrary.mockResolvedValueOnce({ didCancel: true });

    const { getByTestId } = render(<Profile />);

    await act(async () => {
      await fireEvent.press(getByTestId('profile-update-picture'));
    });

    expect(mockToast.show).toHaveBeenCalledTimes(0);
  });

  it('should show error toast when update profile picture failed', async () => {
    launchImageLibrary.mockRejectedValueOnce();

    const { getByTestId } = render(<Profile />);

    await act(async () => {
      await fireEvent.press(getByTestId('profile-update-picture'));
    });

    expect(mockToast.show).toHaveBeenCalledTimes(1);
  });
});
