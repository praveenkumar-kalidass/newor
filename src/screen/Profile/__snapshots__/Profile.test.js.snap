// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "position": "absolute",
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={1334}
          bbWidth={750}
          focusable={false}
          height={1334}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 1334,
                "width": 750,
              },
            ]
          }
          width={750}
        >
          <RNSVGGroup>
            <RNSVGCircle
              cx={0}
              cy={1334}
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              r={250}
              stroke={4278199178}
              strokeWidth={25}
            />
            <RNSVGCircle
              cx={0}
              cy={1334}
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              r={150}
              stroke={4278199178}
              strokeWidth={25}
            />
            <RNSVGCircle
              cx={750}
              cy={0}
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              r={250}
              stroke={4289527808}
              strokeWidth={25}
            />
            <RNSVGCircle
              cx={750}
              cy={0}
              fill={4294967295}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              r={150}
              stroke={4289527808}
              strokeWidth={25}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        flex={1}
        onLayout={[Function]}
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <View
            dataSet={Object {}}
            elevation={1}
            shadowColor="#000000"
            shadowOffset={
              Object {
                "height": 1,
                "width": 0,
              }
            }
            shadowOpacity={0.8}
            shadowRadius={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#eeeeee",
                  "borderRadius": 20,
                  "flex": 1,
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
                Array [
                  Object {},
                ],
              ]
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 40,
                  "width": "100%",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
                styleProps={Object {}}
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#00238a",
                        "color": "#00238a",
                        "marginRight": 8,
                        "paddingRight": 8,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                />
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 20,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "letterSpacing": 0,
                      "lineHeight": 30,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  PROFILE
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ÔÄç
                </Text>
              </View>
            </View>
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="profile-update-picture"
              >
                <View
                  colorScheme="muted"
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "flex-end",
                        "backgroundColor": "#ad0000",
                        "borderColor": "transparent",
                        "borderRadius": 9999,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "marginBottom": -20,
                        "marginRight": -20,
                        "paddingBottom": 2,
                        "paddingLeft": 8,
                        "paddingRight": 8,
                        "paddingTop": 2,
                        "zIndex": 1,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={Object {}}
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#ffffff",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ÔÅÄ
                    </Text>
                  </View>
                </View>
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#ffffff",
                      "borderColor": "#27272a",
                      "borderRadius": 9999,
                      "height": 96,
                      "justifyContent": "center",
                      "position": "relative",
                      "width": 96,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 40,
                        },
                        undefined,
                        Object {
                          "fontFamily": "FontAwesome",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    Ôàõ
                  </Text>
                </View>
              </View>
            </View>
            <View
              dataSet={Object {}}
              style={
                Object {
                  "marginBottom": 20,
                  "marginTop": 20,
                }
              }
            >
              <RCTScrollView
                contentContainerStyle={
                  Array [
                    Object {},
                    Object {
                      "dataSet": Object {},
                    },
                  ]
                }
                data={
                  Array [
                    Object {
                      "key": "firstName",
                      "label": "FIRST_NAME",
                    },
                    Object {
                      "key": "lastName",
                      "label": "LAST_NAME",
                    },
                    Object {
                      "key": "email",
                      "label": "EMAIL",
                    },
                    Object {
                      "key": "mobileNumber",
                      "label": "MOBILE_NUMBER",
                    },
                  ]
                }
                dataSet={Object {}}
                getItem={[Function]}
                getItemCount={[Function]}
                keyExtractor={[Function]}
                onContentSizeChange={[Function]}
                onLayout={[Function]}
                onMomentumScrollBegin={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                style={Object {}}
                viewabilityConfigCallbackPairs={Array []}
              >
                <View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "marginBottom": 4,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        FIRST_NAME
                      </Text>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 16,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                            "lineHeight": 24,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        Hello
                      </Text>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <View
                        aria-orientation="horizontal"
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": "#d4d4d4",
                            "height": 1,
                            "marginBottom": 8,
                            "marginTop": 8,
                            "width": "100%",
                          }
                        }
                        thickness="1"
                      />
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "marginBottom": 4,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LAST_NAME
                      </Text>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 16,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                            "lineHeight": 24,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        World
                      </Text>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <View
                        aria-orientation="horizontal"
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": "#d4d4d4",
                            "height": 1,
                            "marginBottom": 8,
                            "marginTop": 8,
                            "width": "100%",
                          }
                        }
                        thickness="1"
                      />
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "marginBottom": 4,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        EMAIL
                      </Text>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 16,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                            "lineHeight": 24,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        helloworld@test.com
                      </Text>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <View
                        aria-orientation="horizontal"
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": "#d4d4d4",
                            "height": 1,
                            "marginBottom": 8,
                            "marginTop": 8,
                            "width": "100%",
                          }
                        }
                        thickness="1"
                      />
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      dataSet={Object {}}
                      style={Object {}}
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "marginBottom": 4,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        MOBILE_NUMBER
                      </Text>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 16,
                            "fontStyle": "normal",
                            "fontWeight": "400",
                            "letterSpacing": 0,
                            "lineHeight": 24,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        9876543210
                      </Text>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <View
                        aria-orientation="horizontal"
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": "#d4d4d4",
                            "height": 1,
                            "marginBottom": 8,
                            "marginTop": 8,
                            "width": "100%",
                          }
                        }
                        thickness="1"
                      />
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={0}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": -600,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 600,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "flex": 1,
              "justifyContent": "space-between",
              "paddingBottom": 80,
              "paddingTop": 80,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          data={
            Array [
              Object {
                "icon": <Icon
                  allowFontScaling={false}
                  color="#000000"
                  name="power-off"
                  size={24}
                />,
                "id": "LOGOUT",
                "label": "LOGOUT",
                "onPress": [Function],
              },
            ]
          }
          dataSet={Object {}}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={Object {}}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "#7f7f7f",
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 20,
                        "paddingTop": 8,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ÔÄë
                    </Text>
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "width": 12,
                        }
                      }
                    />
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LOGOUT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
